glue/plugins/wcs_autolinking/tests/test_wcs_autolinking.py

```diff
@@ -240,3 +241,254 @@ def test_celestial_with_unknown_axes():
+    px2 = data2.pixel_component_ids
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 1
+    link = links[0]
+    assert isinstance(link, MultiLink)
+    assert len(link) == 2
+    assert ' '.join(str(link[0].get_to_id()).split()[:2]) == ' '.join(str(py1).split()[:2])
+    assert ' '.join(str(link[0].get_from_ids()).split()[:2]) == ' '.join(str(px2).split()[:2])
+
+
+def test_link_of_spectral_axes_of_different_physical_types():
+    """
+    To check that there is no auto-link of spectral axes of two different physical types, e.g.
+    between FREQ and WAVE.
+    TODO: To modify the codebase so that there will be a link between FREQ and WAVE.
+    """
+
+    wcs1 = WCS(naxis=1)
+    wcs1.wcs.ctype = ['FREQ']
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones(2)
+
+    wcs2 = WCS(naxis=1)
+    wcs2.wcs.ctype = ['WAVE']
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones(2)
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 0
+
+
+def test_cube_has_celestial_and_cube_without_celestial_axes_1():
+    """
+    To test that there should be a link between a 3D dataset with celestial axes
+    and a 2D dataset with no celestial axes (variant 1).
+    """
+
+    wcs1 = WCS(naxis=3)
+    wcs1.wcs.ctype = 'RA---TAN', 'FREQ', 'DEC--TAN'
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones((2, 3, 4))
+    pz1, py1, px1 = data1.pixel_component_ids
+
+    wcs2 = WCS(naxis=2)
+    wcs2.wcs.ctype = 'FREQ', 'TIME'
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones((4, 5))
+    py2, px2 = data2.pixel_component_ids
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 1
+    link = links[0]
+    assert isinstance(link, MultiLink)
+    assert len(link) == 2
+    assert link[0].get_to_id() == px2
+    assert link[0].get_from_ids() == [py1]
+    assert link[1].get_to_id() == py1
+    assert link[1].get_from_ids() == [px2]
+
+def test_cube_has_celestial_and_cube_without_celestial_axes_2():
+    """
+    To test that there should be a link between a 3D dataset with celestial axes
+    and a 2D dataset with no celestial axes (variant 2).
+    TODO: To modify code base so that the FREQ axis would be linked up with the WAVE axis.
+    """
+
+    wcs1 = WCS(naxis=3)
+    wcs1.wcs.ctype = 'RA---TAN', 'FREQ', 'DEC--TAN'
+    wcs1.wcs.set()
+
+    data1 = Data(label='Data 1')
+    data1.coords = wcs1
+    data1['x'] = np.ones((2, 3, 4))
+
+    wcs2 = WCS(naxis=2)
+    wcs2.wcs.ctype = 'WAVE', 'TIME'
+    wcs2.wcs.set()
+
+    data2 = Data(label='Data 2')
+    data2.coords = wcs2
+    data2['x'] = np.ones((4, 5))
+
+    dc = DataCollection([data1, data2])
+    links = wcs_autolink(dc)
+    assert len(links) == 0
```

Comment:
As before

Reply:
Sure, am adding the `@pytest.mark.xfail` decorator and changing to 
```python
assert len(links) == 1
```